### PR TITLE
Enable flake8-bugbear B020 lint

### PR DESCRIPTION
**Pull request by dmage**
Fixes part of https://github.com/pytorch/pytorch/issues/106571

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Comment by Skylion007** [comment]
You need to enable the rule for ruff in the pyproject.toml as well

**Comment by dmage** [comment]
@Skylion007 I removed B020 from `tool.ruff.ignore`, and `tool.ruff.select` seems to select all `B` checks. Should I make more changes to pyproject.toml?

**Comment by Skylion007** [comment]
The new errors should make it clear that it is working properly in RUFF now. :)

File: test/test_linalg.py [hunk 1]
```diff
@@ -2413,7 +2413,7 @@ def run_subtest(actual_rank, matrix_size, batches, device, svd_lowrank, **option
                 self.assertEqual(v.mT.matmul(V).det().abs(), torch.ones(batches, device=device, dtype=dtype))
 
         all_batches = [(), (1,), (3,), (2, 3)]
-        for actual_rank, size, all_batches in [
+        for actual_rank, size, all_batches in [  # noqa: B020
```
Comment (dmage) [inline test/test_linalg.py:2416]: I don't have ideas for a better name for `all_batches`, so I added `noqa`.

**Comment by Skylion007** [comment]
You'll also need to sign the CLA

File: torch/_inductor/triton_heuristics.py [hunk 2]
```diff
@@ -536,8 +536,9 @@ def _find_names(obj):
     import inspect
 
     frame = inspect.currentframe()
-    for frame in iter(lambda: frame.f_back, None):  # type: ignore[union-attr]
+    while frame is not None:
         frame.f_locals
+        frame = frame.f_back
```
Comment (dmage) [inline torch/_inductor/triton_heuristics.py:541]: This code is a bit magical for me. I suspect traversing through frames and calling `frame.f_locals` creates new objects that are accessible via `gc.get_referrers`, and my change doesn't break anything. I don't have CUDA to confirm that.

Comment (Skylion007) [reply torch/_inductor/triton_heuristics.py:541]: This looks fine to me, but if anything breaks in this PR, it's likely this code.

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -4853,7 +4853,7 @@ def sample_inputs_put(op_info, device, dtype, requires_grad, **kwargs):
     tgt_gen = (make_arg(size) for size in tgt_sizes)
     idx = make_idx((0,), high=1)
     src = make_arg((0,))
-    for tgt, acc in product(tgt, (True, False)):
+    for tgt, acc in product(tgt_gen, (True, False)):
```
[the remark this answers is not in the document]
Comment (Skylion007) [reply torch/testing/_internal/common_methods_invocations.py:4856]: Great, bugbear actually caught something real. :)

**Review by Skylion007 — APPROVED** [review]
Looks good to me

**Pull request closed without merging**